libbpf-tools/filelife.bpf.c

```diff
@@ -181,9 +177,9 @@ int BPF_KRETPROBE(vfs_unlink_ret)
 
 	/* get full-path */
 	if (full_path && eventp->fname.pathes[0] != '/')
-		bpf_dentry_full_path(eventp->fname.pathes + NAME_MAX, NAME_MAX,
-				MAX_PATH_DEPTH - 1,
-				unlink_event->cwd_dentry,
+		bpf_dentry_full_path(eventp->fname.pathes, NAME_MAX,
+				MAX_PATH_DEPTH,
+				unlink_event->dentry,
 				unlink_event->cwd_vfsmnt,
```

Comment:
Could unlink_event->cwd_vfsmnt be an issue as well? The cwd_vfsmnt at vfs_{open,create} may not be the same one as the mount corresponding to 'unlink_event->dentry', right? Maybe you should get the *current* vfsmnt based on unlink_event->dentry?